src/core/Tree/Tree.tsx

```diff
@@ -0,0 +1,269 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { CommonProps, useTheme } from '../utils';
+import '@itwin/itwinui-css/css/tree.css';
+import cx from 'classnames';
+
+export const TreeContext = React.createContext<
+  | {
+      nodeDepth: number;
+      subNodeIds?: string[];
+      hasSubNodes: boolean;
```

Comment:
I don't think this one is needed in context as it is in `NodeData`

Reply:
Removed `hasSubNodes` from context but I had to add it to `TreeNode` props